osu.Game/Rulesets/UI/Playfield.cs

```diff
@@ -76,6 +82,38 @@ private void load()
 
                 AddInternal(Cursor);
             }
+
+            showPlayfieldArea = config.GetBindable<bool>(OsuSetting.ShowPlayfieldArea);
+            playfieldAreaDimLevel = config.GetBindable<double>(OsuSetting.PlayfieldAreaDimLevel);
+            showPlayfieldArea.ValueChanged += _ => UpdateVisuals();
+            playfieldAreaDimLevel.ValueChanged += _ => UpdateVisuals();
+            UpdateVisuals();
+        }
+
+        protected virtual void UpdateVisuals()
+        {
+            if (playfieldArea == null)
+            {
+                if (showPlayfieldArea.Value)
+                {
+                    AddInternal(playfieldArea = new EditorPlayfieldBorder
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Alpha = (float)playfieldAreaDimLevel.Value,
+                    });
```

Comment:
should not be dynamically added like this, but probably added once at the point of `load()` and just hidden/shown depending on the settings here.